lua/telescope/config.lua

```diff
@@ -279,6 +279,42 @@ local telescope_defaults = {
   ]],
   },
 
+  cache_picker = {
```

Comment:
Rather than having 3 (or more) values could we do something similar to `layout_config` and `history` where we have one table setting with multiple keys?

Reply:
I hope this is fine now; I'm afraid this increased control flow complexity quite a bit as I'm writing to the `pickers.cache_picker` during caching and `opts.cache_picker = false` can be validly passed to `builtin.{resume, pickers}`.

Reply:
I like the config now. I am more concerned about a better user interface than a couple more ifs in code :laughing: 
Also i love this table config to group stuff since clason proposed it for history. way better than having multiple string values that are prefixed with something